pkg/utils/utils.go

```diff
@@ -61,3 +65,59 @@ func SendToSocket(socket string, command string) error {
 	}
 	return nil
 }
+
+// checkValidity runs a HAProxy configuration validity check on a file
+func checkValidity(configFile string) error {
+	out, err := exec.Command("haproxy", "-c", "-f", configFile).CombinedOutput()
+	if err != nil {
+		glog.Warningf("Error validating config file:\n%v", string(out))
+		return err
+	}
+	return nil
+}
+
+// multibinderERBOnly generates a config file from ERB template by invoking multibinder-haproxy-erb
+func multibinderERBOnly(configFile string) (string, error) {
+	_, err := exec.Command("multibinder-haproxy-erb", "/usr/local/sbin/haproxy", "-f", configFile, "--erb-write-only").CombinedOutput()
+	if err != nil {
+		return "", err
+	}
+	return configFile[:strings.LastIndex(configFile, ".erb")], nil
+}
+
+// RewriteConfigFiles safely replaces configuration files with new contents after validation
+func RewriteConfigFiles(data []byte, reloadStrategy, configFile string) error {
+	tmpf := "/etc/haproxy/new_cfg.erb"
```

Comment:
I was considering this, but placing the temporary file in the same directory as the target file ensures the `os.Rename` will perform on the same filesystem and be atomic so that seemed like a safer bet.

I also considered using `ioutil.TempFile` as a safest option, but that seems a hassle as multibinder expects a filename ending with `.erb` and this is not a feature of `ioutil.TempFile` yet. 

Actually we could use `ioutil.TempDir` created within `/etc/haproxy/` to ensure the same filesystem and no interference with the new file even when the temporary filename itself is hardcoded, at the cost of having to clean up the temporary directory.

Let me know it that sounds good or you prefer something else.